refsheet.cls

```diff
@@ -1,57 +1,212 @@
+%   Disable this option for the body of the document.
+%
+\DeclareStringOption[3]{rscols}
+\AtBeginDocument{%
+  \DisableKeyvalOption[action=error,package=refsheet]{refsheet}{rscols}}
+%
+%   Define a preset margin for geometry
+%
+\DeclareStringOption[1cm]{margin}
+%
+%   Warn about options which will not be used/ passed on because of potential clashes
+%
+\DeclareVoidOption{portrait}{%
+  \PackageWarning{refsheet}{Incompatible with portrait mode}}
+\DeclareVoidOption{twocolumn}{%
+  \PackageError{refsheet}{Using 'multicol' for columns}}
+\DeclareVoidOption{titlepage}{%
+  \PackageWarning{refsheet}{Incompatible with a title page}}
+%
+%   Pass all other options to article
+%
+\DeclareDefaultOption{%
+  \PassOptionsToClass{\CurrentOptionKey}{article}}
+%
+%   DEV: Space for more options
+%
+
+%
+%   Process these options
+%   
+\ProcessKeyvalOptions{refsheet}
+%
+%%%
+
+%%% INHERIT STANDARD CLASS
+%
+%   Latest update of standard class article
+%   (It probably also works with earlier versions, but why sweat it.)
+\LoadClass{article}[2014/09/29]
+%
+%%%
+
+%%% ADDITIONAL REQUIRED TOOLS
+%
+%   Latest update of the nag package. 
+%   Harasses the user about deprecated usages.
+%
+\RequirePackage[orthodox,l2tabu,abort]{nag}[2011/11/25] 
+%
+%   Package etoolbox is necessary for additional hooks like \AtEndPreamble
+%
+\RequirePackage{etoolbox}
+%
+%   Page layout with latest version of geometry
+%
+\RequirePackage{geometry}[2018/04/16]
+\AtEndPreamble{%
+  \geometry{landscape,margin=\refsheet@margin}}
+%
+%   Produce columns with multicol
+%
+\RequirePackage{multicol}[2018/04/20]
+\AfterEndPreamble{%
+  \begin{multicols}{\refsheet@rscols}}
+\AtEndDocument{%
+  \end{multicols}}
+%
+%%%
+
+%%% REDEFINE TITLE
+%
+\renewcommand{\maketitle}{%
+  {%
+   \begin{center}%
+     \Large \@title \\%
+     \vspace{0.1ex}%
+     \small \@author, \@date\end{center}%
+  }%
+}
+%
+%%%
 
-\ProcessOptions
-
-\LoadClass{article}[2013/02/25]
-
-\RequirePackage{fixltx2e}
-\RequirePackageWithOptions{nag}[2013/03/25 othodox,l2tabu,abort]
-
-%% Page layout
-
-\RequirePackage{geometry}[2013/03/25]
-%\usepackage[landscape,margin=0.5in]{geometry}
-\geometry{landscape,margin=0.5in}
-\RequirePackage{multicol}
-
+%%% DOCUMENT DIVISIONS
+%
+%   Package titlesec for setting spacing
+%
+\RequirePackage{titlesec}
+%
+%   Package nopageno so there are no page numbers
+%   (to keep \section from resetting page style).
+%
+\RequirePackage{nopageno} 
+%
+%   No section numbering
+%
+\setcounter{secnumdepth}{0}
+%
+%   No indentation and superfluous spacing
+%
+\setlength{\parindent}{0pt} 
+\titlespacing{\section}{0pt}{0pt}{0pt}
+\titlespacing{\subsection}{0pt}{0pt}{0pt}
+%
+%%%
+
+%%% COMMANDS AND ENVIRONMENTS
+%
+%   Extended definition of lists with enumitem
+%
+\RequirePackage[inline]{enumitem}[2011/08/28]
+%   Calculating widths, e.g. with \widthof
+\RequirePackage{calc} 
+
+% generate dense lists
+\setlist{noitemsep,topsep=0.05ex,}
```

Comment:
Why the extra `,`?

Reply:
I first wanted to include `leftmargin=*` for all lists, but then noticed that won't work well. So I split this for itemised and description lists, see next two lines. I forgot to remove the comma.